src/dmd/utils.d

```diff
@@ -210,6 +140,64 @@ void escapePath(OutBuffer* buf, const(char)* fname)
     }
 }
 
+/**
+ * Takes a path, and make it compatible with Makefile format.
+ *
+ * That is, turns
+ *  - '\' into '/'
+ *  - '$' into "$$"
+ *  - ' ' into "\ "
+ *  - '%' into "\%"
+ *  - '#' into "\#"
+ *  - ':' into "\:"
+ *
+ * Params:
+ *   buf = Buffer to write the escaped path to
+ *   fname = Path to escape
+ */
+void escapeMakePath(OutBuffer *buf, const(char)* fname)
+{
+    while(*fname)
+    {
+        switch (*fname)
+        {
+            case '\\':
+                buf.writeByte('/');
+                break;
+            case '$':
+                buf.writestring("$$");
+                break;
+            case ' ':
+                buf.writestring("\\ ");
```

Comment:
GNU make uses a weird quoting scheme for white space. A space or tab preceded by 2N+1 backslashes represents N backslashes followed by space; a space or tab preceded by 2N backslashes represents N backslashes at the end of a file name; and backslashes in other contexts should not be doubled.

Reply:
FOAD:
```
case ' ':
case '\t':
    for (auto p = fname - 1; orig_fname <= p && *p == '\\';  p--)
        buf.writeByte('\\');
    buf.writeByte('\\');
```
Though the above code is easier to follow if you just had a counter for each backslash you encounter, and:
```
case ' ':
case '\t':
    while (slashes--)
        buf.writeByte('\\');
    buf.writeByte('\\');
```
Resetting the counter if any other character is seen.